addons/account/account.py

```diff
@@ -241,6 +241,14 @@ def _default_outbound_payment_methods(self):
     profit_account_id = fields.Many2one('account.account', string='Profit Account', domain=[('deprecated', '=', False)], help="Used to register a profit when the ending balance of a cash register differs from what the system computes")
     loss_account_id = fields.Many2one('account.account', string='Loss Account', domain=[('deprecated', '=', False)], help="Used to register a loss when the ending balance of a cash register differs from what the system computes")
 
+    bank_account_id = fields.Many2one('res.partner.bank', ondelete='restrict')
+    bank_acc_number = fields.Char('Account Number', related='bank_account_id.acc_number')
```

Comment:
Remove these fields from the journal: bank_acc_number, bank_acc_number_type, bank_id, bank_name, bank_bic.
Instead of putting these fields on the journal, just show the bank_account_id as required. (they will create the bank information from the bank form view). The reasons:
   - some l10n_* module may extend the bank form view (like BIC, ...)
   - it's ok for the user to fill in a many2one if default values are provided in the context

Keep: display_on_footer (that's not related to all bank accounts but only the company one), bank,account_id